lib/arelastic/queries/bool.rb

```diff
@@ -35,8 +35,9 @@ def merge_expression(clause, expr, klass:, merge_test_method:)
         mergeable, other = expr.partition { |el| el.is_a?(klass) }
         merged = mergeable.group_by(&merge_test_method).flat_map do |_, group|
           if group.size > 1
-            group.first.instance_variable_set("@query", self.class.new(clause => group.map(&:query)))
-            group.first
+            item = group.first.dup
+            item.instance_variable_set("@query", self.class.new(clause => group.map(&:query)))
+            item
```

Comment:
I don't love this `instance_variable_set`. Really we should probably just be rebuilding the node from scratch. But this is how it is now, and it's an easy patch - maybe I'll follow up with a separate PR